### DISCUSSION TITLE
Does Gordon support ANDROIDX_TEST_ORCHESTRATOR?

### DISCUSSION THREAD

**Discussion by gorru** [Banno/Gordon#106] [Q&A] [answered]
Hi all.
I didn't find any reference to this information.
I noticed che trying Gordon most of our expresso test faild due to process crash.
This is not what we have running normal connectedFullDebugAndroidTest gradle task.
So I thought that it could depend on using the orchestrator or not.

Could you help me?
Thanks

**Comment by joshschriever** [comment]
Gordon works independently of test orchestrator - they're separate, incompatible ways of running tests. But I don't think having test orchestrator enabled should affect Gordon one way or another.

You might try disabling test orchestrator and then running `connectedFullDebugAndroidTest`. If you get the process crash failures then, there must be something wrong with your app or your tests, and not something specific to Gordon. If that succeeds (meaning your failures are specific to Gordon) then I'd probably need more information about your setup in order to investigate.

**Comment by gorru** [comment]
Thanks.
My first doubt is that the android orchestrator hides some wrong behaviour
that our tests have. These behaviours are obviously our mistake.
The other doubt is about gordon behaviour related to test independence. I
mean: the android orchestrator runs each test independently and clears the
app data folder before each test.
Does Gordon do something similar to prevent application status incoherences
during tests execution?

**Comment by joshschriever** [comment, marked as answer]
Gordon does not do anything to clear data between tests.
See also:
https://github.com/Banno/Gordon/issues/20#issuecomment-576750815
https://github.com/Banno/Gordon/discussions/2#discussioncomment-205645

**Comment by gorru** [comment]
Thank you very much!
